File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/support/DocWriteResponse.java
package fr.myprysm.vertx.elasticsearch.action.support;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.elasticsearch.action.DocWriteResponse.Result;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
@DataObject(generateConverter = true)
public class DocWriteResponse extends ReplicationResponse {

    private ShardId shardId;

    private String index;
    private String type;
    private String id;

    private long version;
    private long seqNo;
    private long primaryTerm;
    private boolean forcedRefresh;
    private Result result;

    public DocWriteResponse(DocWriteResponse other) {
        super(other);
        shardId = other.shardId;
        index = other.index;
        type = other.type;
        id = other.id;
        version = other.version;
        seqNo = other.seqNo;
        primaryTerm = other.primaryTerm;
        forcedRefresh = other.forcedRefresh;
        result = other.result;
    }

    public DocWriteResponse(ReplicationResponse other) {
        super(other);
    }

    public DocWriteResponse(JsonObject json) {
        super(json);
        DocWriteResponseConverter.fromJson(json, this);
    }

    public JsonObject toJson() {
        JsonObject json = super.toJson();
        DocWriteResponseConverter.toJson(this, json);
        return json;
    }

}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/support/DocWriteResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.support

import fr.myprysm.vertx.elasticsearch.action.support.DocWriteResponse
import fr.myprysm.vertx.elasticsearch.action.support.ShardId
import fr.myprysm.vertx.elasticsearch.action.support.ShardInfo
import org.elasticsearch.action.DocWriteResponse.Result

fun DocWriteResponse(
        forcedRefresh: Boolean? = null,
        id: String? = null,
        index: String? = null,
        primaryTerm: Long? = null,
        result: Result? = null,
        seqNo: Long? = null,
        shardId: fr.myprysm.vertx.elasticsearch.action.support.ShardId? = null,
        shardInfo: fr.myprysm.vertx.elasticsearch.action.support.ShardInfo? = null,
        type: String? = null,
        version: Long? = null): DocWriteResponse = fr.myprysm.vertx.elasticsearch.action.support.DocWriteResponse().apply {

    if (forcedRefresh != null) {
        this.setForcedRefresh(forcedRefresh)
    }
    if (id != null) {
        this.setId(id)
    }
    if (index != null) {
        this.setIndex(index)
    }
    if (primaryTerm != null) {
        this.setPrimaryTerm(primaryTerm)
    }
    if (result != null) {
        this.setResult(result)
    }
    if (seqNo != null) {
        this.setSeqNo(seqNo)
    }
    if (shardId != null) {
        this.setShardId(shardId)
    }
    if (shardInfo != null) {
        this.setShardInfo(shardInfo)
    }
    if (type != null) {
        this.setType(type)
    }
    if (version != null) {
        this.setVersion(version)
    }
}
